project/SparkBuild.scala

```diff
@@ -1141,7 +1141,8 @@ object TestSettings {
     (Test / javaOptions) += "-ea",
     (Test / javaOptions) ++= {
       val metaspaceSize = sys.env.get("METASPACE_SIZE").getOrElse("1300m")
-      val extraTestJavaArgs = Array("-XX:+IgnoreUnrecognizedVMOptions",
```

Comment:
what's the actual change?

Reply:
The actual change is to expose an ENV variable to set the heap size. If set this heap size will be using as `-Xmx` while invoking `runMain` from sbt. 
The default 4g headspace was not sufficient when I was running the newly added benchmark for 10 billion rows (driver OOM'd), thus the change.

While making the change I noticed `extraTestJavaArgs` `Array` was first converted to a string and immediately at the next line it was again split on spaces and converted to `Seq`.
Thus I refactored the code a bit.